src/kioskboard.js

```diff
@@ -573,12 +573,17 @@
 
           // keys click listeners: begin
           var keysClickListeners = function (input) {
+            var addKeyClickListener = function (keyElm, clickHandler) {
+              keyElm.addEventListener("contextmenu", function (e) { e.preventDefault(); }, false);
+              keyElm.addEventListener("click", clickHandler);
+              keyElm.addEventListener("touchend", clickHandler);
+            };
```

Comment:
Hi @surexxx ,

Could you please move that function with some comments and with some name changes to the one level up(scope) => (Line 573)

_(In addition, we are using single quotes.)_

```js

// keys event listeners: begin
var keysEventListeners = function (keyElement, onClickHandler) {
  if (keyElement) {
    keyElement.addEventListener('contextmenu', function (event) {
      event.preventDefault();
    }, false);
    keyElement.addEventListener('click', onClickHandler);
    keyElement.addEventListener('touchend', onClickHandler);
  }
};
// keys event listeners: end

```

Thanks,
Furkan.

Reply:
Hi, @surexxx ,

And also I am considering not having two event handlers(`click` and `touchend`) on touchable devices...

So,

```js

// keys event listeners: begin
var keysEventListeners = function (keyElement, onClickHandler) {
  if (keyElement) {
    var isTouchableDevice = 'ontouchend' in window || window.navigator.maxTouchPoints > 0;

    if (isTouchableDevice) {
      keyElement.addEventListener('contextmenu', function (event) {
        event.preventDefault();
      }, false);
      keyElement.addEventListener('touchend', onClickHandler);
    } else {
      keyElement.addEventListener('click', onClickHandler);
    }
  }
};
// keys event listeners: end

```

WDT?

Reply:
Unfortunately there is a bug on this code fragement. We tested on a touchable laptop which also supports "click", and 'click' doesn't work, only 'touchend'.
```javascript

                    // keys event listeners: begin
                    var keysEventListeners = function (keyElement, onClickHandler) {
                        if (!keyElement) {
                            return;
                        }

                        var isTouchableDevice = 'ontouchend' in window || window.navigator.maxTouchPoints > 0;

                        if (isTouchableDevice) {
                            keyElement.addEventListener('contextmenu', function (event) {
                                event.preventDefault();
                            }, false);
                            keyElement.addEventListener('touchend', onClickHandler);
                        }

                        keyElement.addEventListener('click', onClickHandler);                            
                    };
```
@furcan would you like to fix it? I can fix it also....

Reply:
Hi @surexxx ,

The point is, `click` or `touchend` will be going to do the exact same things at the end. I already suggested separating the events. Because otherwise, it will be a bug, doing all the stuff(onClickHandler) at the same time with two different events for the same purposes.

If your device is already supporting the `touchend`, there is no need to use the `click` event at the same time in our case. It will be worked whether it is a long press or not.

Reply:
@furcan thanks a lot :)